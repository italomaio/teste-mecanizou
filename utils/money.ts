export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
