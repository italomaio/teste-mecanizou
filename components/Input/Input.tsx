import * as React from "react";

import { cn } from "@/utils/classes";

const Input = React.forwardRef<HTMLInputElement, React.ComponentProps<"input">>(
  ({ className, type, ...props }, ref) => {
    return (
      <input
        role="textbox"
        type={type}
        ref={ref}
        className={cn(
          "flex h-9 w-full rounded-sm bg-transparent px-3 py-1 text-base shadow-sm transition-colors placeholder:text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50 md:text-sm",
          "focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-sapphire text-gray-700 dark:text-zinc-100 border border-sapphire",
          "file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground",
          className
        )}
        {...props}
      />
    );
  }
);
Input.displayName = "Input";

export default Input;
