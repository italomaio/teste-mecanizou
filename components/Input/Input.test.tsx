import React from "react";

import { fireEvent, screen } from "@testing-library/react";
import { render } from "@/tests/utils";

import Input from "./Input";

describe("Input Component", () => {
  it("Should render the input element", () => {
    render(<Input />);
    const inputElement = screen.getByRole("textbox");
    expect(inputElement).toBeInTheDocument();
  });

  it("Should apply custom className", () => {
    render(<Input className="custom-class" />);
    const inputElement = screen.getByRole("textbox");
    expect(inputElement).toHaveClass("custom-class");
  });

  it("Should forward the ref to the input element", () => {
    const ref = React.createRef<HTMLInputElement>();
    render(<Input ref={ref} />);
    expect(ref.current).toBeInstanceOf(HTMLInputElement);
  });

  it("Should handle type prop correctly", () => {
    render(<Input type="password" />);
    const inputElement = screen.getByRole("textbox");
    expect(inputElement).toHaveAttribute("type", "password");
  });

  it("Should disable when the disabled prop is set", () => {
    render(<Input disabled />);
    const inputElement = screen.getByRole("textbox");
    expect(inputElement).toBeDisabled();
  });

  it("Should render placeholder text", () => {
    render(<Input placeholder="Enter text" />);
    const inputElement = screen.getByPlaceholderText("Enter text");
    expect(inputElement).toBeInTheDocument();
  });

  it("Should call onChange handler when typing with correct value", () => {
    const handleChange = jest.fn();
    render(<Input onChange={handleChange} />);
    const inputElement = screen.getByRole("textbox");
    inputElement.focus();

    fireEvent.change(inputElement, { target: { value: "test" } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it("Should call onFocus handler when focused", () => {
    const handleFocus = jest.fn();
    render(<Input onFocus={handleFocus} />);
    const inputElement = screen.getByRole("textbox");
    inputElement.focus();
    expect(handleFocus).toHaveBeenCalledTimes(1);
  });

  it("Should call onBlur handler when blurred", () => {
    const handleBlur = jest.fn();
    render(<Input onBlur={handleBlur} />);
    const inputElement = screen.getByRole("textbox");
    inputElement.focus();
    inputElement.blur();
    expect(handleBlur).toHaveBeenCalledTimes(1);
  });

  it("Should match the snapshot", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});
