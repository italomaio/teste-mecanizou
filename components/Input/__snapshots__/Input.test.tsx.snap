// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Input Component Should match the snapshot 1`] = `
<DocumentFragment>
  <input
    class="flex h-9 w-full rounded-sm bg-transparent px-3 py-1 text-base shadow-sm transition-colors placeholder:text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-sapphire text-gray-700 dark:text-zinc-100 border border-sapphire file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground"
    role="textbox"
  />
</DocumentFragment>
`;
